vendor/github.com/docker/go-connections/tlsconfig/config.go

```diff
@@ -83,6 +88,65 @@ func certPool(caFile string) (*x509.CertPool, error) {
 	return certPool, nil
 }
 
+func IsErrEncryptedKey(err error) bool {
+	return errors.Cause(err) == errCantDecrypt
+}
+
+// getPrivateKey returns the private key in 'keyBytes', in PEM-encoded format.
+// If the private key is encrypted, 'passphrase' is used to decrypted the
+// private key.
+func getPrivateKey(keyBytes []byte, passphrase string) ([]byte, error) {
+	// this section makes some small changes to code from notary/tuf/utils/x509.go
+	pemBlock, _ := pem.Decode(keyBytes)
+	if pemBlock == nil {
+		return nil, fmt.Errorf("no valid private key found")
+	}
+
+	var err error
+	if x509.IsEncryptedPEMBlock(pemBlock) {
+		keyBytes, err = x509.DecryptPEMBlock(pemBlock, []byte(passphrase))
+		if err != nil {
+			return nil, errCantDecrypt
```

Comment:
Thanks, just updated the code. I will open a PR on go-connections for the tlsconfig/config.go (perhaps I need to add some unit tests as part of that PR for the new functionality).